fpsd/features.py

```diff
@@ -0,0 +1,768 @@
+        greater than x
+
+        Args:
+            lengths [list of int]: number of lengths to create bins.
+            Bin limits are from the given length to the end of the trace
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self.drop_table("features.burst_windowed_lengths")
+
+        feature_columns = ["num_bursts_with_length_gt_{}".format(length)
+                           for length in lengths]
+
+        # Use LEFT OUTER JOIN because many of the later windows will be Null
+        # Note: count(*) will return Null if count(*) = 0 hence coalesce is
+        # used to set those cells to 0
+        subqueries = ["""LEFT OUTER JOIN (SELECT exampleid,
+                         COALESCE(count(burst), 0) AS {colname}
+                         FROM public.current_bursts WHERE
+                         burst > {length} GROUP BY exampleid)
+                         t{table_ref} ON foo.exampleid =
+                         t{table_ref}.exampleid""".format(colname=feature_columns[feat_ind],
+                                                        length=length,
+                                                        table_ref=feat_ind)
+                      for feat_ind, length in enumerate(lengths)]
+
+        query = """CREATE TABLE features.burst_windowed_lengths AS (
+                   SELECT foo.exampleid, {} FROM (
+                   (SELECT exampleid, count(*) FROM
+                   raw.frontpage_traces GROUP BY exampleid) foo
+                   {} ));""".format(", ".join(feature_columns),
+                                    " ".join(subqueries))
+
+        self.engine.execute(query)
+        return "features.burst_windowed_lengths"
+
+    def create_table_burst_lengths(self, num_bursts=100):
+        """This method takes all bursts and produces a table with the
+        following format:
+
+        exampleid  | a         | b         ...    | num_bursts
+        (integer)  | (bigint)  | (bigint)  ...    | (bigint)
+        --------------------------------------------------------
+        9          | 1         | 2         ...
+        10         | 2         | 2         ...
+        11         | 1         | 3         ...
+
+        where there are a variable number (num_bursts) of feature
+        columns, each named length_burst_x. For example, length_burst_1
+        is the length of the first burst.
+
+        Args:
+            num_bursts [int]: number of bursts to make features for
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self.drop_table("features.burst_lengths")
+
+        column_names = ['length_burst_{} bigint'.format(x)
+                        for x in range(1, num_bursts + 1)]
+
+        query = """CREATE TABLE features.burst_lengths AS
+                   (SELECT * FROM crosstab(
+                   'SELECT exampleid, rank, burst
+                   FROM public.current_bursts')
```

Comment:
You need to add an `ORDER BY RANK` directive to this inner, crosstab query, and update the last method of the last `TestCase` accordingly.